applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#ifndef _AMOD_API_H_
+#define _AMOD_API_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include "data_fifo.h"
+#include "aobj_api.h"
+
+/**
+ * @brief Modules private handle.
+ */
+struct handle;
+
+/**
+ * @brief Private context for the module.
+ */
+struct amod_context;
+
+/**
+ * @brief Modules private configuration.
+ */
+struct amod_configuration;
+
+/**
+ * @brief Module type relative to the module.
+ */
+enum amod_type {
```

Comment:
I think it would be best if the enums were over the structs (meaning over struct handle)